src/cmake/config.cmake.in

```diff
@@ -0,0 +1,13 @@
+include(CMakeFindDependencyMacro)
+
+find_library(D3D12_LIB d3d12.lib REQUIRED)
```

Comment:
What if someone wants to load D3D12 dynamically using `LoadLibrary`, `GetProcAddress` rather than statically linking to these libraries? I wouldn't require them as a dependency, rather let the user link/load them themselves.

Reply:
Hi, @adam-sawicki-a, let me explain your demands, the unrelated files is easy, i can move them to next PRs, the config.cmake.in, if you wish, is okay to remove the dependencies, for the final result, in addition of the result files in the installation, because this PR is only for the cmake installation process, in the install folder will be created another folder, in share, with the D3D12MA config and target files, (each `-` indicates one level dir):
`
- share
-- cmake
--- D3D12MemoryAllocator
---- D3D12MemoryAllocatorConfig.cmake
---- D3D12MemoryAllocatorConfigVersion.cmake
---- D3D12MemoryAllocatorTargets.cmake
---- D3D12MemoryAllocatorTargets-debug.cmake
`

With these files, is possible, in cmake to use the command: 
`
find_package(D3D12MemoryAllocator)
#linkage method
target_link_libraries(${TARGET_NAME} PUBLIC GPUOpen::D3D12MemoryAllocator)
`

This principle is used in VMA, but is a little different because the lib is header only.